### PR TITLE
CHE-2869: upgrade to GWT 2.8

### PR DESCRIPTION
**Pull request by svor**
#

### What does this PR do?
Upgrade to GWT 2.8

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/2869

#### Changelog

Changelog: Upgraded GWT to version 2.8 including new `guava` and `lib-gwt-svg`.

#### Release Notes

n/a

#### Docs
n/a

### CODE REVIEW

File: assembly/onpremises-ide-compiling-war-ide/pom.xml [hunk 1]
```diff
@@ -422,12 +422,12 @@
                     </dependency>
                 </dependencies>
                 <configuration>
+                    <gwtSdkFirstInClasspath>true</gwtSdkFirstInClasspath>
```
Comment (skabashnyuk) [inline assembly/onpremises-ide-compiling-war-ide/pom.xml]: Can you describe what is it?

File: assembly/onpremises-ide-compiling-war-ide/pom.xml [hunk 2]
```diff
@@ -27,8 +27,8 @@
     <name>Onpremises IDE :: WAR :: Compiling App</name>
     <properties>
         <generated.sources.directory>${project.build.directory}/generated-sources/gen</generated.sources.directory>
-        <maven.compiler.source>1.7</maven.compiler.source>
-        <maven.compiler.target>1.7</maven.compiler.target>
+        <maven.compiler.source>1.8</maven.compiler.source>
+        <maven.compiler.target>1.8</maven.compiler.target>
```
Comment (skabashnyuk) [inline assembly/onpremises-ide-compiling-war-ide/pom.xml]: create single variable. Or reuse it from che variable

File: plugins/plugin-hosted/codenvy-hosted-ext-factory/pom.xml [hunk 3]
```diff
@@ -27,8 +27,8 @@
     <name>Codenvy Plugin :: Hosted :: Factory</name>
     <properties>
         <dto-generator-out-directory>${project.build.directory}/generated-sources/dto/</dto-generator-out-directory>
-        <maven.compiler.source>1.7</maven.compiler.source>
-        <maven.compiler.target>1.7</maven.compiler.target>
+        <maven.compiler.source>1.8</maven.compiler.source>
+        <maven.compiler.target>1.8</maven.compiler.target>
```
Comment (skabashnyuk) [inline plugins/plugin-hosted/codenvy-hosted-ext-factory/pom.xml]: variable

File: plugins/plugin-hosted/codenvy-hosted-ext-hosted/pom.xml [hunk 4]
```diff
@@ -26,8 +26,8 @@
     <packaging>jar</packaging>
     <name>Codenvy Plugin :: Hosted Environment Tuning</name>
     <properties>
-        <maven.compiler.source>1.7</maven.compiler.source>
-        <maven.compiler.target>1.7</maven.compiler.target>
+        <maven.compiler.source>1.8</maven.compiler.source>
+        <maven.compiler.target>1.8</maven.compiler.target>
```
Comment (skabashnyuk) [inline plugins/plugin-hosted/codenvy-hosted-ext-hosted/pom.xml]: variable

File: plugins/plugin-microsoft/codenvy-plugin-microsoft-vsts-ext-client/pom.xml [hunk 5]
```diff
@@ -26,8 +26,8 @@
     <name>Codenvy Plugin :: Microsoft :: VSTS Client</name>
     <properties>
         <dto-generator-out-directory>${project.build.directory}/generated-sources/dto/</dto-generator-out-directory>
-        <maven.compiler.source>1.7</maven.compiler.source>
-        <maven.compiler.target>1.7</maven.compiler.target>
+        <maven.compiler.source>1.8</maven.compiler.source>
+        <maven.compiler.target>1.8</maven.compiler.target>
```
Comment (skabashnyuk) [inline plugins/plugin-microsoft/codenvy-plugin-microsoft-vsts-ext-client/pom.xml]: variable

File: plugins/plugin-pullrequest/codenvy-plugin-pullrequest-ide/pom.xml [hunk 6]
```diff
@@ -25,8 +25,8 @@
     <artifactId>codenvy-plugin-pullrequest-ide</artifactId>
     <name>Codenvy Plugin :: Pull request :: IDE</name>
     <properties>
-        <maven.compiler.source>1.7</maven.compiler.source>
-        <maven.compiler.target>1.7</maven.compiler.target>
+        <maven.compiler.source>1.8</maven.compiler.source>
+        <maven.compiler.target>1.8</maven.compiler.target>
```
Comment (skabashnyuk) [inline plugins/plugin-pullrequest/codenvy-plugin-pullrequest-ide/pom.xml]: variable

File: wsmaster/codenvy-hosted-machine-authentication-ide/pom.xml [hunk 7]
```diff
@@ -27,8 +27,8 @@
     <properties>
         <dto-generator-out-directory>${project.build.directory}/generated-sources/dto/</dto-generator-out-directory>
         <findbugs.failonerror>false</findbugs.failonerror>
-        <maven.compiler.source>1.7</maven.compiler.source>
-        <maven.compiler.target>1.7</maven.compiler.target>
+        <maven.compiler.source>1.8</maven.compiler.source>
+        <maven.compiler.target>1.8</maven.compiler.target>
```
Comment (skabashnyuk) [inline wsmaster/codenvy-hosted-machine-authentication-ide/pom.xml]: variable

File: assembly/onpremises-ide-compiling-war-ide/pom.xml (hunk 1 above)
Comment (svor) [inline assembly/onpremises-ide-compiling-war-ide/pom.xml]: This is necessary for the correct classpath construction, we need to use  gwt sdk jars first in classpath.

File: assembly/onpremises-ide-compiling-war-ide/pom.xml (hunk 2 above)
Comment (svor) [inline assembly/onpremises-ide-compiling-war-ide/pom.xml]: I removed these properties. They will be used from [che-parent](https://github.com/eclipse/che-parent/blob/master/pom.xml#L121)

**Comment by skabashnyuk** [comment]
please update excludes here https://github.com/eclipse/che-archetypes too

**Comment by svor** [comment]
> please update excludes here https://github.com/eclipse/che-archetypes too

PR --> https://github.com/eclipse/che-archetypes/pull/12

**Pull request merged**